controlplane/kubeadm/internal/workload_cluster_coredns_test.go

```diff
@@ -287,62 +291,75 @@ kind: ClusterConfiguration
 	// Namespace/Name (e.g. The CoderDNS ConfigMap & Deployment, the kubeadm ConfigMap), it is not possible to run the use cases in parallel.
 	for _, tt := range tests {
 		g := NewWithT(t)
-		t.Log(tt.name)
-
-		for _, o := range tt.objs {
-			// NB. deep copy test object so changes applied during a test does not affect other tests.
-			o := o.DeepCopyObject().(client.Object)
-			g.Expect(testEnv.Create(ctx, o)).To(Succeed())
-		}
 
-		w := &Workload{
-			Client:          testEnv.GetClient(),
-			CoreDNSMigrator: tt.migrator,
-		}
-		err := w.UpdateCoreDNS(ctx, tt.kcp)
-		if tt.expectErr {
-			g.Expect(err).To(HaveOccurred())
-			return
-		}
-		g.Expect(err).ToNot(HaveOccurred())
+		t.Run(tt.name, func(t *testing.T) {
+			for _, o := range tt.objs {
+				// NB. deep copy test object so changes applied during a test does not affect other tests.
+				o := o.DeepCopyObject().(client.Object)
+				g.Expect(testEnv.Create(ctx, o)).To(Succeed())
+				// this makes sure that the cache is updated with the object
+				// to avoid 404 errors leading to test flakes
+				g.Eventually(func() bool {
```

Comment:
It _might_ be faster if we create everything in one for loop, and then use a second for loop to verify presence in the cache?

Reply:
I was gonna look into @fabriziopandini 's suggestion and mimic the `Eventually` assertion in the `testEnv.Create()` method and remove this part altogether.

Reply:
I'm +1 to keep everything in a single for loop, given that my major concern about tests is readability/maintainability, and I consider speed a secondary concern.